### PR TITLE
Manual rotate

### LINKED ISSUE #100: Is there a logrotate sink that supports archive existing log file on launch
Currently it only appends to existing file and archive on reaching max size. Is it possible to add  an option to archive existing on launch?

### PR DESCRIPTION
**Pull request by xgdgsc**
resolves #100 , I have no idea how to do unit testing for this. If you give some clue I would like to add Tests.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -142,7 +142,8 @@ int main(int argc, char**argv) {
    // The call is asynchronously executed on your custom sink.
    const int k10MBInBytes = 10 * 1024 * 1024;
    std::future<void> received = sinkHandle->call(&LogRotate::setMaxLogSize, k10MBInBytes);
-   
+   // call this if you want to rotate log in program manually
```
Comment (KjellKod) [inline README.md:145]: Nitpicking: maybe consider changing the sentence for a better flow
Possibly something like:
` You can manually trigger a rotate of the log `

File: logrotate/test/RotateFileTest.cpp [hunk 2]
```diff
@@ -1,304 +1,325 @@
...
+  EXPECT_TRUE(DoesFileEntityExist(logfilename));
+  logrotate.rotateLog();
+  logrotate.save("test2");
+  ASSERT_FALSE(checkIfExist("test1")) << "\n\tcontent:" << content;
+  ASSERT_TRUE(checkIfExist("test2")) << "\n\tcontent:" << content;
+  auto app_name = _filename + ".log";
+  logrotate.rotateLog();
+  logrotate.save("test3");
+  auto allFiles =
+      LogRotateUtility::getLogFilesInDirectory(_directory, app_name);
```
Comment (KjellKod) [inline logrotate/test/RotateFileTest.cpp:323]: This looks good. We aren't testing the number of gz files existence but **I think that's fine.** 
I.e. there is no need to look into the benign time-race condition where  the `test1` `gz` file will be overwritten by the `test1`  - unless a new second period has started.

**Comment by KjellKod** [comment]
Once the formatting issue is taken care of this will be fine to merge. Thank you for the effort!

**Comment by xgdgsc** [comment]
Sorry, I tried to save every time without format but the editor sometimes saves automatically. Do you have a clang-format file or style that I can use to format it?

**Comment by KjellKod** [comment]
[SublimeAstyleFormatter.sublime-settings.zip](https://github.com/KjellKod/g3sinks/files/4848944/SublimeAstyleFormatter.sublime-settings.zip)

Here's the SublimeText3  format I use through the package SublimeAStyleFormatter. 
Alternatively, you can just hand fix the changes.  It's actually pretty easy. 

Copy out your changes.  revert the file. Copy back your change. 
Obviously you need to fix your editor so it doesn't auto format the code upon saving. Not sure what editor you use but AFAIK all of the ones I have tried can turn/on-off auto format.

**Comment by KjellKod** [comment]
I fixed the most glaring formatting differences. the rest I can live with. Please merge this PR, into your PR and this work should be good to go.   https://github.com/xgdgsc/g3sinks/pull/1

**Comment by KjellKod** [comment]
@xgdgsc ^

**Pull request merged**